src/js/angular/clustermanagement/services/cluster-drawing.service.js

```diff
@@ -177,68 +180,149 @@ function getNodeInfoIconType(node) {
 
     switch (node.recoveryStatus.state) {
         case RecoveryState.SEARCHING_FOR_NODE:
```

Comment:
For both functions you can construct an object before the switch and just set the `icon` value. Then return the object. It would be more readable IMO.